src/compiler/scala/tools/nsc/backend/jvm/opt/InlinerHeuristics.scala

```diff
@@ -64,7 +64,7 @@ class InlinerHeuristics[BT <: BTypes](val bTypes: BT) {
     }).filterNot(_._2.isEmpty).toMap
   }
 
-  private def isTraitStaticSuperAccessorName(s: String) = s.endsWith("$")
+  private def isTraitStaticSuperAccessorName(s: String) = s != "$init$" && s.endsWith("$")
```

Comment:
... it might be more robust to look for a static method named `foo$(self)` that only calls `self.foo`.